pkg/cloudprovider/provider/vsphere/helper.go

```diff
@@ -69,15 +69,26 @@ func createClonedVM(ctx context.Context, vmName string, config *Config, session
 		targetVMFolder = datacenterFolders.VmFolder
 	}
 
-	datastore, err := session.Finder.Datastore(ctx, config.Datastore)
+	relocateSpec := types.VirtualMachineRelocateSpec{
```

Comment:
As the comment below suggests, we split the cloning mechanism from reconfiguring the newly spawned vm due to some permissions validating. I think this is a good approach to have better insights into what went wrong.

Reply:
Personally I'm not convinced that splitting clone, relocate, reconfigure in three distinct steps is a good idea. We could do this with one single API call and we would reduce the probability of failing in the middle of the machine preparation. In addition if the issue is that the error message is not explicit enough we should address it elsewhere. That said I did that because in order to have a placement recommendation I had to provide the `CloneSpec` that include that include the `VirtualMachineRelocateSpec`. At this point cloning the VM and relocate it after that seemed a nonsense to me.

Reply:
I agree with Iacopo. The previous authors never properly determined and documented the necessary permissions, so "tell which specific action failed due to lacking permissions" was used as a crutch for trial-and-error.